tests/jsunit/connection_db_test.js

```diff
@@ -93,3 +93,163 @@ function test_DB_removeConnection() {
+  assertEquals(result.length, 0);
+
+  // Test block has no neighbours due to being out of range diagonally
+  result = helper_getNeighbours(db, -2, -2, 2);
+  assertEquals(result.length, 0);
+}
+
+function test_DB_findPositionForConnection() {
+    var db = new Blockly.ConnectionDB();
+    db.addConnection_(helper_createConnection(0, 0, Blockly.PREVIOUS_STATEMENT));
+    db.addConnection_(helper_createConnection(0, 1, Blockly.PREVIOUS_STATEMENT));
+    db.addConnection_(helper_createConnection(0, 2, Blockly.PREVIOUS_STATEMENT));
+    db.addConnection_(helper_createConnection(0, 4, Blockly.PREVIOUS_STATEMENT));
+    db.addConnection_(helper_createConnection(0, 5, Blockly.PREVIOUS_STATEMENT));
+
+    assertEquals(5, db.length);
+    var conn = helper_createConnection(0, 3, Blockly.PREVIOUS_STATEMENT);
+    assertEquals(3, db.findPositionForConnection_(conn));
+}
+
+function test_DB_findConnection() {
+    var db = new Blockly.ConnectionDB();
+    for (var i = 0; i < 10; i++) {
+      db.addConnection_(helper_createConnection(i, 0, Blockly.PREVIOUS_STATEMENT));
+      db.addConnection_(helper_createConnection(0, i, Blockly.PREVIOUS_STATEMENT));
+    }
+
+    var conn = helper_createConnection(3, 3, Blockly.PREVIOUS_STATEMENT);
+    db.addConnection_(conn);
+    assertEquals(conn, db[db.findConnection(conn)]);
+
+    conn = helper_createConnection(3, 3, Blockly.PREVIOUS_STATEMENT);
+    assertEquals(-1, db.findConnection(conn));
+}
+
+function test_DB_ordering() {
+    var db = new Blockly.ConnectionDB();
+    for (var i = 0; i < 10; i++) {
+        db.addConnection_(helper_createConnection(0, 9 - i, Blockly.PREVIOUS_STATEMENT));
+    }
+
+    for (i = 0; i < 10; i++) {
+        assertEquals(i, db[i].y_);
+    }
+
+    // quasi-random
+    var xCoords = [-29, -47, -77, 2, 43, 34, -59, -52, -90, -36, -91, 38, 87, -20, 60, 4, -57,
+            65, -37, -81, 57, 58, -96, 1, 67, -79, 34, 93, -90, -99, -62, 4, 11, -36, -51, -72,
+            3, -50, -24, -45, -92, -38, 37, 24, -47, -73, 79, -20, 99, 43, -10, -87, 19, 35,
+            -62, -36, 49, 86, -24, -47, -89, 33, -44, 25, -73, -91, 85, 6, 0, 89, -94, 36, -35,
+            84, -9, 96, -21, 52, 10, -95, 7, -67, -70, 62, 9, -40, -95, -9, -94, 55, 57, -96,
+            55, 8, -48, -57, -87, 81, 23, 65];
+    var yCoords = [-81, 82, 5, 47, 30, 57, -12, 28, 38, 92, -25, -20, 23, -51, 73, -90, 8, 28,
+            -51, -15, 81, -60, -6, -16, 77, -62, -42, -24, 35, 95, -46, -7, 61, -16, 14, 91, 57,
+            -38, 27, -39, 92, 47, -98, 11, -33, -72, 64, 38, -64, -88, -35, -59, -76, -94, 45,
+            -25, -100, -95, 63, -97, 45, 98, 99, 34, 27, 52, -18, -45, 66, -32, -38, 70, -73,
+            -23, 5, -2, -13, -9, 48, 74, -97, -11, 35, -79, -16, -77, 83, -57, -53, 35, -44,
+            100, -27, -15, 5, 39, 33, -19, -20, -95];
+    for (i = 0; i < xCoords.length; i++) {
+        db.addConnection_(helper_createConnection(xCoords[i], yCoords[i],
+          Blockly.PREVIOUS_STATEMENT));
```

Comment:
2 space indent for levels, 4 space indent for wrapping.  This whole file needs some attention regarding indents.